launch_testing/launch_testing/proc_info_handler.py

```diff
@@ -109,6 +109,35 @@ def assertWaitForShutdown(self,
         success = False
 
         def proc_is_shutdown():
+            try:
+                resolved_process = resolveProcesses(
+                    info_obj=self._proc_info_handler,
+                    process=process,
+                    cmd_args=cmd_args,
+                    strict_proc_matching=True
+                )[0]
+
+                process_event = self._proc_info_handler[resolved_process]
+                return hasattr(process_event, 'returncode')
```

Comment:
@pbaughman why not checking explicitly for process exit events?

Reply:
Because when I write Python, I forget that types are a thing!  I like your idea - it's more explicit.